src/lib/components/SubscribeForm/index.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * @fileoverview Element that renders newsletter subscription form.
+ */
+
+import {html} from "lit-element";
+import {BaseElement} from "../BaseElement";
+import {countries} from "./countries";
+
+/**
+ * Element that renders newsletter subscription form.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class SubscribeForm extends BaseElement {
+  static get properties() {
+    return {
+      errors: {
+        type: Object,
+      },
+      submitted: {
+        type: Boolean,
+      },
+    };
+  }
+
+  constructor() {
+    super();
+    this.checkboxes = ["WebDevNewsletter", "collects-pii-spii-checkbox"];
+    this.needsDoubleOptIn = [
+      "AT: Austria",
+      "DE: Germany",
+      "GR: Greece",
+      "LU: Luxembourg",
+      "NO: Norway",
+    ];
+    this.robotName = "is-it-just-me-or-was-this-form-filled-out-by-a-robot";
+    this.submissionUrl =
+      "https://services.google.com/fb/submissions/591768a1-61a6-4f16-8e3c-adf1661539da/";
+    this.submitted = false;
+  }
+
+  /**
+   * @param {FormData} form
+   * @return {FormData}
+   */
+  cleanForm(form) {
+    const country = form.get("Country");
+    this.checkboxes.forEach((box) => {
+      const optedIn = this.needsDoubleOptIn.includes(country)
+        ? "Unconfirmed"
+        : "True";
+      form.set(box, optedIn);
+    });
+
+    form.append("LanguagePreference", "en-US");
+    form.delete(this.robotName);
+
+    return form;
+  }
+
+  submit(form) {
+    return new Promise((resolve, reject) => {
+      const request = new XMLHttpRequest();
+      request.open("POST", this.submissionUrl);
+      request.send(form);
+      request.onload = function() {
+        const response = request.response ? JSON.parse(request.response) : {};
+
+        if (response && response.result === "accepted") {
+          resolve();
+        } else if (response && response.errors) {
+          reject(response.errors);
+        } else {
+          reject("Could not submit, please try again.");
+        }
+      };
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    if (this.submitted === false) {
+      const form = new FormData(e.target);
+      const formIsRobot = form.get(this.robotName).length !== 0;
+
+      if (!formIsRobot) {
+        const cleanedForm = this.cleanForm(form);
+        this.submit(cleanedForm)
+          .then(() => (this.submitted = true))
+          .catch((e) => (this.errors = e));
+      } else {
+        this.submitted = true;
+      }
+    }
+
+    return false;
+  }
+
+  renderForm() {
+    let errorMessage;
+
+    if (this.errors) {
+      if (typeof this.errors === "string") {
+        errorMessage = html`
+          <div class="web-subscribe-error">
+            <p>${this.errors}</p>
+          </div>
+        `;
+      } else {
+        const errorMessages = Object.values(this.errors).map(
```

Comment:
Isn't `this.errors` going to contain a single `Error`? This makes it seem like it's an `Object` containing many. But on line 91 you're just setting one.

Reply:
So when an invalid submission is made you'll receive an object with a key for each field that's invalid. I've refactored it to take the object and get all the errors out of it. This may change so that instead of listing all the errors in the bottom of the form to show the error message for the field that was invalid.